### DISCUSSION TITLE
Why MediatR?

### DISCUSSION THREAD

**Discussion by opflucker** [jasontaylordev/CleanArchitecture#1009] [General]
It is widly discussed that MediatR is neither a mediator pattern implementation nor a best practice. Some reads if anyone are interested in this:

-   https://arialdomartini.github.io/mediatr
-   https://cezarypiatek.github.io/post/why-i-dont-use-mediatr-for-cqrs

Why put this in a template?

**Comment by apolonas** [comment]
I have been using this template with MediatR for 3 years now in various projects. It has helped me and my teams tremendously. We take advantage of the MediatR behaviors to add custom logic around commands and queries and even utilize it appropriately in combination with the Strategy pattern to isolate custom logic for different clients within the same solution. I have used it in APIs and Function Apps and it works great.

**Reply by opflucker** [reply]
I have been in varios projects, in different client companies, along the last 12 years. Most of them worked with MediatR, and as you share, developers were convinced it was helpful. In all cases when I can show them how it could be done without it, they agree it is a lot more simple but reject to abandom MediatR. They prefer continue with the extra complexity "just in case it becomes useful".

Because most of the projects I worked on does not get any benefit using MediatR, but the opposite, my experience match with those articules I shared. I think MediatR is a bad practice by default. Of course, maybe you work with a kind of projects when that extra complexity layer is really helpful.

**Reply by apolonas** [reply]
I believe any project can benefit from  Aspect Oriented Programming and the decorator pattern. That’s what mediatR behaviors are for. But hey to each his own!

**Reply by DennisJansenDev** [reply]
> I have been in varios projects, in different client companies, along the last 12 years. Most of them worked with MediatR, and as you share, developers were convinced it was helpful. In all cases when I can show them how it could be done without it, they agree it is a lot more simple but reject to abandom MediatR. They prefer continue with the extra complexity "just in case it becomes useful".
> 
> 
> 
> Because most of the projects I worked on does not get any benefit using MediatR, but the opposite, my experience match with those articules I shared. I think MediatR is a bad practice by default. Of course, maybe you work with a kind of projects when that extra complexity layer is really helpful.
> 
> 

Lmfao what a nonsense, if you find mediatr adds complexity to your project, maybe you should consider switching professions.

**Reply by opflucker** [reply]
@DennisJansenDev , if you do not undestand what complexity means in software, maybe you should go to school again, or consider switch your profession.

**Reply by opflucker** [reply]
@apolonas , I agree with you about the decorator element in MeadiatR.

**Reply by DennisJansenDev** [reply]
> @DennisJansenDev , if you do not undestand what complexity means in software, maybe you should go to school again, or consider switch your profession.

Hahahaha do everyone a favor and stop writing software, please.

**Comment by punkouter26** [comment]
this is an outrage!!
________________________________
